sql/2020/22_HTTP_2/number_of_http_sites_returning_upgrade_http_header_containing_h2.sql

```diff
@@ -0,0 +1,30 @@
+# standardSQL
+# Number of HTTP (not HTTPS) sites which return upgrade HTTP header containing h2.
+CREATE TEMPORARY FUNCTION getUpgradeHeader(payload STRING)
+RETURNS STRING
+LANGUAGE js AS """
+  try {
+    var $ = JSON.parse(payload);
+    var headers = $.response.headers;
+    var st = headers.find(function(e) { 
+      return e['name'].toLowerCase() === 'upgrade'
+    });
+    return st['value'];
+  } catch (e) {
+    return '';
+  }
+""";
+
+SELECT 
+  client,
+  firstHtml,  
+  COUNT(0) AS num_requests
+FROM 
+  `httparchive.almanac.requests`
+WHERE
+  date='2020-08-01' AND 
+  url LIKE "http://%" AND
+  getUpgradeHeader(payload) LIKE "%h2%"
+GROUP BY
+  client,
+  firstHtml
```

Comment:
This should be in the `WHERE` clause so that you're only analyzing the response for the document.

Reply:
@rviscomi I am not following. You are stating that THIS should be in the `Where` clause....Please define THIS

Reply:
`firstHtml`

Reply:
Made the changes and tested successfully

Reply:
@rviscomi is suggesting only to measure this on the initial page and not all requests. In which case you should remove from `SELECT` and `GROUP BY` clause and add to `WHERE` clause (no point in being in `SELECT` or `GROUP BY` if always true).

I'm not sure I agree to be honest. It may be interesting to see if ANY request does this and not just document itself. And with the group by we get to choose whatever stat we want. So think I'd leave it as is to be honest. I came up with this stat last year so I'm using my "I know more than Rick on this one (for once!)" card! 😀

At end of the day it was only on [108 sites last year](https://almanac.httparchive.org/en/2019/http2#http2-issues) so probably not worth including this year unless that's got a lot worse. But let's run as is to have a look.

We should however rename this file from "number of sites..." to "number of requests..." which is probably why Rick suggested this change.

Reply:
Leaving firstHtml in Select and Group By and renamed the filename to reflect requests instead of sites